livekit-plugins/livekit-plugins-google/livekit/plugins/google/multimodal/live_api.py

```diff
@@ -0,0 +1,353 @@
+    Voice,
+    _build_tools,
+)
+
+
+@dataclass
+class GeminiContent(Content):
+    pass
+
+
+@dataclass
+class ModelOptions:
+    model: MultimodalModels | str
+    api_key: str | None
+    voice: Voice | str
+    response_modalities: ResponseModality
+    vertexai: bool
+    project: str | None
+    location: str | None
+    candidate_count: int
+    temperature: float | None
+    max_output_tokens: int | None
+    top_p: float | None
+    top_k: int | None
+    presence_penalty: float | None
+    frequency_penalty: float | None
+    instructions: str
+
+
+class RealtimeModel(MultimodalModel):
+    def __init__(
+        self,
+        *,
+        instructions: str = "",
+        model: MultimodalModels | str = "gemini-2.0-flash-exp",
+        api_key: str | None = None,
+        voice: Voice | str = "Puck",
+        response_modalities: ResponseModality = "AUDIO",
+        vertexai: bool = False,
+        project: str | None = None,
+        location: str | None = None,
+        candidate_count: int = 1,
+        temperature: float | None = None,
+        max_output_tokens: int | None = None,
+        top_p: float | None = None,
+        top_k: int | None = None,
+        presence_penalty: float | None = None,
+        frequency_penalty: float | None = None,
+        loop: asyncio.AbstractEventLoop | None = None,
+    ):
+        self._model = model
+        self._loop = loop or asyncio.get_event_loop()
+        self._api_key = api_key or os.environ.get("GOOGLE_API_KEY")
+        self._vertexai = vertexai
+        self._project_id = project or os.environ.get("GOOGLE_PROJECT")
+        self._location = location or os.environ.get("GOOGLE_LOCATION")
+        if self._api_key is None and not self._vertexai:
+            raise ValueError("GOOGLE_API_KEY is not set")
+
+        self._rt_sessions: list[GeminiRealtimeSession] = []
+        self._opts = ModelOptions(
+            model=model,
+            api_key=api_key,
+            voice=voice,
+            response_modalities=response_modalities,
+            vertexai=vertexai,
+            project=project,
+            location=location,
+            candidate_count=candidate_count,
+            temperature=temperature,
+            max_output_tokens=max_output_tokens,
+            top_p=top_p,
+            top_k=top_k,
+            presence_penalty=presence_penalty,
+            frequency_penalty=frequency_penalty,
+            instructions=instructions,
+        )
+
+    @property
+    def sessions(self) -> list[GeminiRealtimeSession]:
+        return self._rt_sessions
+
+    def session(
+        self,
+        *,
+        chat_ctx: llm.ChatContext | None = None,
+        fnc_ctx: llm.FunctionContext | None = None,
+    ) -> MultimodalSession:
+        session = GeminiRealtimeSession(
+            opts=self._opts,
+            chat_ctx=chat_ctx or llm.ChatContext(),
+            fnc_ctx=fnc_ctx,
+            loop=self._loop,
+        )
+        self._rt_sessions.append(session)
```

Comment:
as per my understanding, we are maintaining list of sessions because we could've multiple sessions running, they will get closed at once whenever `aclose` is called. @theomonnom correct me if I'm wrong

Reply:
it is a WeakSet